add accessibility guidelines to docs style guide (#77145)
ansible/ansible: docs/docsite/rst/dev_guide/style_guide/index.rst

Every page should have at least one anchor to support internal ``:ref:`` links.
Long pages, or pages with multiple levels of headers, can also include a local TOC.

.. note::

> 	Avoid raw URLs. RST and sphinx allow ::code:`https;//my.example.com`, but this is unhelpful for those using screen readers. ``:ref:`` links automatically pick up the header from the anchor, but for external links, always use the ::code:`link title <link-url>`_` format.

Reviewer: ```suggestion
	Avoid raw URLs. RST and sphinx allow ::code:`https://my.example.com`, but this is unhelpful for those using screen readers. ``:ref:`` links automatically pick up the header from the anchor, but for external links, always use the ::code:`link title <link-url>`_` format.
```

   .. contents::
      :local:

Accessibility guidelines
=========================

Ansible documentation has a goal to be more accessible. Use the following guidelines to help us reach this goal.

Images and alternative text

>   Ensure all icons, images, diagrams, and non text elements have a meaningful alternative-text description. Do no include screen caps of CLI output. use ``code-block`` instead.

Reviewer: ```suggestion
  Ensure all icons, images, diagrams, and non text elements have a meaningful alternative-text description. Do not include screen captures of CLI output. Use ``code-block`` instead.
```
I think we want to fully say "capture" instead of "cap" to make this clearer for people who may not be familiar with this?

Tables
  Tables have a simple, logical reading order from left to right, and top to bottom.
  Tables include a header row and avoid empty or blank table cells.
  Label tables with a descriptive title.

  .. code-block:: reStructuredText

    .. table:: File descriptions

      +----------+----------------------------+
      |File      |Purpose                     |
      +==========+============================+
      |foo.txt   |foo configuration settings  |
      +----------+----------------------------+
      |bar.txt   |bar configuration settings  |
      +----------+----------------------------+

Colors and other visual information
  * Avoid instructions that rely solely on sensory characteristics. For example, do not use ``Click the square, blue button to continue.``
  * Convey information by methods and not by color alone.
  * Ensure there is sufficient contrast between foreground and background text or graphical elements in images and diagrams.
  * Instructions for navigating through an interface make sense without directional indicators such as left, right, above, and below.

>   * Nest headers in order from H1 to H2, H3, and so forth, without skipping heading levels. See :ref:`headers_style` for the correct order.

Reviewer 2: Isn't this done automatically by RST/Spinx?

Author: just tested and you are correct. It spits out a CRITICAL warning if the headers are out of order so to speak.
